docs/src/types.md

```diff
@@ -0,0 +1,44 @@
+#LightGraph Types
+
+The LightGraph library supports both the `AbstractGraph` type and two concrete simple graph types- - `Graph` for undirected graphs and `DiGraph` for directed graphs -- that are subtypes of `AbstractGraph`.
+
+## SimpleGraph Types
+
+*LightGraphs.jl* provides two concrete graph types (referred to as "simple graphs"): `Graph` is an undirected graph, and `DiGraph` is its directed counterpart. Both of these types can be parameterized to specifying how vertices are identified (by default, `Graph` and `DiGraph` use the system default integer type, usually `Int64`).
```

Comment:
Technically, `Graph` and `DiGraph` are aliases for `SimpleGraph` and `SimpleDiGraph` used to provide backwards compatibility.

Reply:
@sbromberger this has definitely always confused me a little. Is the best to just say `Graph` and `DiGraph` and not bring the "simple" component into it?

Reply:
Yeah, it's confusing to me too. At some point what I'd like to do is get rid of `Graph` and `DiGraph` as type aliases and maybe do what Plots does with its backend: that is, load a graph type (or types) and then let the user set a default.

Reply:
Yeah that is a little tricky, I am in favor of encouraging people to use `Graph/DiGraph` names and then if they want to change to a different concrete implementation they can do

```julia
using LightGraphs
using SimpleWeightedGraphs
import SimpleWeightedGraphs: SimpleWeightedGraph

Graph = SimpleWeightedGraph

g = Graph(n)
...
```